### PR TITLE
Package format added as argument to the getCnaName method to temporary skip pypi format packages

### LINKED ISSUE #21235: The python packages generate false positives due to its lack of vendor
## Description

The alerts generated and not generated by both the legacy vulnerability detector and the refactored one were compared. See https://github.com/wazuh/wazuh/issues/21012 for more details.

It was found that the collected python packages don't have the vendor field and are scanned against the NVD.
But sometimes, these packages have the same name as some other ones and their small version

- CVE-2011-0729: **language-selector**, `0.1`  (https://ubuntu.com/security/CVE-2011-0729 https://nvd.nist.gov/vuln/detail/ CVE-2011-0729). The python package **language-selector** (`/usr/lib/python3/dist-packages/language_selector-0.1.egg-info/PKG-INFO`) doesn't have a vendor, and it matches in the NVD `MATCH Package: language-selector, Version: 0.1, CVE: CVE-2011-0729, Ver: 0, RLT: , RLTE: 0.6.6`. But the package is in fact `language-selector-common` `0.211`. The legacy module doesn't seem to be scanning the python packages. Same as CVE-2015-1330 for `unattended-upgrades`

This issue can be fixed in many ways, and all of them should be considered. More of one could be applied:
- Content fix: if the packages wouldn't have such a big vulnerable range, these real small version wouldn't match
- Scanner/feedManager and CNA: the NVD might not have vulnerabilities for python packages so we should avoid this situation. It should be investigated if we could have the same situation in the vendor feed, and if the platform filter could help
- Skip python packages: maybe the python packages collected by **syscollector** aren't useful for the vulnerability scanner without a specific feed and should be avoided for the moment

## DoD

- [x] Analyze the situation and proposals. Discuss with the team.
- [x] Implement the changes.
- [x] Add UT
- [x] CI Pass

### PR DESCRIPTION
**Pull request by matias-braida**
|Related issue|
|---|
|Closes #21235  |

## Description

This PR modified the method TDatabaseFeedManager::getCnaName adding the packet format argument to allow the CNA decision algorithm to take it into account. For the particular case of PiPy format, temporary it return no CNA string. Also the PackageScanner logic was modified to allow skip scan of packages without CNA specified.
UTs were also added.

## Logs and Evidence
Previous to the changes the PyPi package "language-selector" was scanned against nvd feed and found as vulnerable.
Here are the previous logs:

```
2024/01/16 13:15:56 wazuh-modulesd:vulnerability-scanner[41942] packageScanner.hpp:53 at handleRequest(): DEBUG: Initiating a vulnerability scan for package 'language-selector' on Agent '78e8dc8901e2' (ID: '001', Version: 'v4.8.0').
2024/01/16 13:15:56 wazuh-modulesd:vulnerability-scanner[41942] packageScanner.hpp:191 at handleRequest(): DEBUG: Initiating a vulnerability scan for package 'language-selector' with CVE Numbering Authorities (CNA) 'nvd' on Agent '78e8dc8901e2' (ID: '001', Version: 'v4.8.0').
2024/01/16 13:15:56 wazuh-modulesd:vulnerability-scanner[41942] packageScanner.hpp:77 at operator()(): DEBUG: Scanning package - 'language-selector' (Installed Version: 0.1, Security Vulnerability: CVE-2011-0729). Identified vulnerability: Version: 0. Required Version Threshold: . Required Version Threshold (or Equal): 0.6.6.
2024/01/16 13:15:56 wazuh-modulesd:vulnerability-scanner[41942] packageScanner.hpp:129 at operator()(): INFO: Match found, the package 'language-selector', is vulnerable to 'CVE-2011-0729'. Current version: '0.1' (less than '' or equal to '0.6.6'). - Agent '78e8dc8901e2' (ID: '001', Version: 'v4.8.0').
2024/01/16 13:15:56 wazuh-modulesd:vulnerability-scanner[41942] packageScanner.hpp:77 at operator()(): DEBUG: Scanning package - 'language-selector' (Installed Version: 0.1, Security Vulnerability: CVE-2011-1842). Identified vulnerability: Version: 0. Required Version Threshold: . Required Version Threshold (or Equal): 0.6.6.
2024/01/16 13:15:56 wazuh-modulesd:vulnerability-scanner[41942] packageScanner.hpp:129 at operator()(): INFO: Match found, the package 'language-selector', is vulnerable to 'CVE-2011-1842'. Current version: '0.1' (less than '' or equal to '0.6.6'). - Agent '78e8dc8901e2' (ID: '001', Version: 'v4.8.0').
2024/01/16 13:15:56 wazuh-modulesd:vulnerability-scanner[41942] packageScanner.hpp:77 at operator()(): DEBUG: Scanning package - 'language-selector' (Installed Version: 0.1, Security Vulnerability: CVE-2013-1066). Identified vulnerability: Version: 0.110. Required Version Threshold: . Required Version Threshold (or Equal): .
2024/01/16 13:15:56 wazuh-modulesd:vulnerability-scanner[41942] packageScanner.hpp:77 at operator()(): DEBUG: Scanning package - 'language-selector' (Installed Version: 0.1, Security Vulnerability: CVE-2013-1066). Identified vulnerability: Version: 0.79. Required Version Threshold: . Required Version Threshold (or Equal): .
2024/01/16 13:15:56 wazuh-modulesd:vulnerability-scanner[41942] packageScanner.hpp:77 at operator()(): DEBUG: Scanning package - 'language-selector' (Installed Version: 0.1, Security Vulnerability: CVE-2013-1066). Identified vulnerability: Version: 0.79.1. Required Version Threshold: . Required Version Threshold (or Equal): .
2024/01/16 13:15:56 wazuh-modulesd:vulnerability-scanner[41942] packageScanner.hpp:77 at operator()(): DEBUG: Scanning package - 'language-selector' (Installed Version: 0.1, Security Vulnerability: CVE-2013-1066). Identified vulnerability: Version: 0.79.2. Required Version Threshold: . Required Version Threshold (or Equal): .
2024/01/16 13:15:56 wazuh-modulesd:vulnerability-scanner[41942] packageScanner.hpp:77 at operator()(): DEBUG: Scanning package - 'language-selector' (Installed Version: 0.1, Security Vulnerability: CVE-2013-1066). Identified vulnerability: Version: 0.79.3. Required Version Threshold: . Required Version Threshold (or Equal): .
2024/01/16 13:15:56 wazuh-modulesd:vulnerability-scanner[41942] packageScanner.hpp:77 at operator()(): DEBUG: Scanning package - 'language-selector' (Installed Version: 0.1, Security Vulnerability: CVE-2013-1066). Identified vulnerability: Version: 0.90. Required Version Threshold: . Required Version Threshold (or Equal): .
2024/01/16 13:15:56 wazuh-modulesd:vulnerability-scanner[41942] packageScanner.hpp:215 at handleRequest(): DEBUG: Vulnerability scan for package 'language-selector' on Agent '001' has completed.
```

After the modification, the same package is skip since its format is PyPi.
```
2024/01/16 16:58:38 wazuh-modulesd:vulnerability-scanner[85308] packageScanner.hpp:53 at handleRequest(): DEBUG: Initiating a vulnerability scan for package 'language-selector' on Agent '78e8dc8901e2' (ID: '001', Version: 'v4.8.0').
2024/01/16 16:58:38 wazuh-modulesd:vulnerability-scanner[85308] packageScanner.hpp:194 at handleRequest(): DEBUG: Vulnerability scan skipped because CNA can't be resolved for package 'language-selector' on Agent '78e8dc8901e2' (ID: '001', Version: 'v4.8.0'). Vendor: ' '. Format: 'pypi'
```

## Tests

- Compilation without warnings in every supported platform
  - [ ] Linux
- [ ] Run UTs successfully
- [ ] Run Manual tests

### CODE REVIEW

File: src/wazuh_modules/vulnerability_scanner/src/databaseFeedManager/databaseFeedManager.hpp [hunk 1]
```diff
@@ -547,10 +547,16 @@ class TDatabaseFeedManager final : public Observer<nlohmann::json&>
     /**
      * @brief Get CNA/ADP name based on the package vendor.
      * @param vendor Package vendor.
+     * @param format Package format.
      * @return CNA/ADP name.
      */
-    std::string getCnaName(std::string_view vendor)
+    std::string getCnaName(std::string_view vendor, std::string_view format)
     {
+        if (!format.compare("pypi"))
```
Comment (pereyra-m) [inline src/wazuh_modules/vulnerability_scanner/src/databaseFeedManager/databaseFeedManager.hpp:555]: I'd add a comment to explain this temporary workaround.

File: src/wazuh_modules/vulnerability_scanner/src/databaseFeedManager/databaseFeedManager.hpp [hunk 2]
```diff
@@ -547,10 +547,16 @@ class TDatabaseFeedManager final : public Observer<nlohmann::json&>
     /**
      * @brief Get CNA/ADP name based on the package vendor.
```
Comment (pereyra-m) [inline src/wazuh_modules/vulnerability_scanner/src/databaseFeedManager/databaseFeedManager.hpp:548]: Please, update the description of the method: `... based on the package vendor`

**Review by pereyra-m — APPROVED** [review]
Gj!

Just an optional comment.

File: src/wazuh_modules/vulnerability_scanner/src/databaseFeedManager/databaseFeedManager.hpp (hunk 1 above)
Comment (matias-braida) [reply src/wazuh_modules/vulnerability_scanner/src/databaseFeedManager/databaseFeedManager.hpp:555]: done

**Review by pereyra-m — APPROVED** [review]

**Pull request merged**